### ISSUE TITLE
Benefits of export_ir vs export_op?

### ISSUE THREAD

**Issue by kmn1024** [OpenVINO-dev-contest/llama2.openvino#9]
I would like to convert a new model (Phi2), with quantization. I see several paths, but am not sure what the benefits are of each:
1. Change export_ir.py and export the model. Run quantize.py.
2. Run export_op.py (as is) to export the model. Run quantize.py.
3. GPTQ the model myself. Run export_op.py on it (does running export_ir.py here make any difference)?

Which path is likely to produce the highest quality model? How about the fastest inference?

**Comment by OpenVINO-dev-contest** [comment]
Hi @kmn1024 for inference performance, it depends on how much percentage of your weight is compressed to int4 precision. 
In most case, all these path can bring you almost same performance, but I believe the accuracy of GPTQ model is the best.

**Comment by kmn1024** [comment]
Thanks for the quick response! OK, to simulate converting my own model, I did the following:
1. cd ~/models; git lfs clone https://huggingface.co/TheBloke/Llama-2-7B-Chat-GPTQ/
2. Setup llama2.openvino
3. python3 export_op.py -m ~/models/Llama-2-7B-Chat-GPTQ/
> INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, onnx, openvino
 --- exporting IR --- 
CUDA extension not installed.
CUDA extension not installed.
Framework not specified. Using pt to export to ONNX.
Using the export variant default. Available variants are:
    - default: The default ONNX variant.
Using framework PyTorch: 2.0.1+cpu
Overriding 1 configuration item(s)
        - use_cache -> True
/root/openvino_env/lib/python3.10/site-packages/transformers/modeling_attn_mask_utils.py:114: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if (input_shape[-1] > 1 or self.sliding_window is not None) and self.is_causal:
/root/openvino_env/lib/python3.10/site-packages/optimum/exporters/onnx/model_patcher.py:299: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if past_key_values_length > 0:
/root/openvino_env/lib/python3.10/site-packages/nncf/torch/dynamic_graph/wrappers.py:75: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  op1 = operator(*args, **kwargs)
/root/openvino_env/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py:150: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if seq_len > self.max_seq_len_cached:
/root/openvino_env/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py:415: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if attn_weights.size() != (bsz, self.num_heads, q_len, kv_seq_len):
/root/openvino_env/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py:422: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if attention_mask.size() != (bsz, 1, q_len, kv_seq_len):
/root/openvino_env/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py:433: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if attn_output.size() != (bsz, self.num_heads, q_len, self.head_dim):
Provided model does not contain state. It may lead to sub-optimal performance.Please reexport model with updated OpenVINO version >= 2023.3.0 calling the `from_pretrained` method with original model and `export=True` parameter
 --- exporting fp16 model --- 
 --- exporting tokenizer --- 

4. python3 pipeline/generate_op.py -m "./ir_model" -p "Einstein said" -d "CPU"                                                                                                                                                                                                                                                        
> INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, onnx, openvino                                                                                             
 --- load tokenizer ---                                                                                                                                                                                                                                                                                                                                                             
 --- use local model ---                                                                                                                                                                  
Provided model does not contain state. It may lead to sub-optimal performance.Please reexport model with updated OpenVINO version >= 2023.3.0 calling the `from_pretrained` method with original model and `export=True` parameter                                                                                                                                                  
Compiling the model to CPU ...                                                                                                                                                            
Setting OpenVINO CACHE_DIR to ir_model/model_cache                                                                                                                                                                                                                                                                                                                                  
 --- start generating ---                                                                                                                                                                 
 --- text decoding ---                                                                                                                                                                                                                                                                                                                                                              
Generation of 128 tokens took 82.747 s on CPU                                                                                                                                                                                                                                                                                                                                       
Response: Einstein said that time is relative, and that it can be affected by gravity.                                                                                                                                                                                                                                                                                              
I think that's a pretty cool idea, but I'm not sure I fully understand it. Can you explain it to me in a way that makes sense?                                                                                                                                                                                                                                                      
Of course! Time is a fundamental concept in physics that describes the duration of events and the order in which they occur. According to Einstein's theory of relativity, time is not absolute, but rather is affected by the strength of the gravitational field it is in.                                                                                                        
In other words, time can slow down or speed up depending on the gravitational pull

Next, I repeated the steps with another quant:
5. git clone --branch gptq-4bit-32g-actorder_True --single-branch https://huggingface.co/TheBloke/Llama-2-7b-Chat-GPTQ.git gptq-4bit-32g-actorder_True
6. ... rm -rf ir_model, repeat export_op.py, repeat pipeline/generate_op.py
> INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, onnx, openvino 
 --- load tokenizer --- 
 --- use local model --- 
Provided model does not contain state. It may lead to sub-optimal performance.Please reexport model with updated OpenVINO version >= 2023.3.0 calling the `from_pretrained` method with original model and `export=True` parameter
Compiling the model to CPU ...
Setting OpenVINO CACHE_DIR to ir_model/model_cache
 --- start generating --- 
 --- text decoding --- 
Generation of 128 tokens took 96.684 s on CPU
Response: Einstein said Rica Rica Rica Rica & & & & & & & &,
 &




Why is the output of gptq-4bit-32g-actorder_True so much worse? Based on the [params explanation](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GPTQ#provided-files-and-gptq-parameters), gptq-4bit-32g-actorder_True should actually be better?

**Comment by OpenVINO-dev-contest** [comment]
Hi @kmn1024 could you share the HF repo of gptq-4bit-32g-actorder_True? Is it a GPQT version of standard llama2, cuz some customized operation may not be supported in OpenVINO runtime.
